src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int maxFigureSize;
+    private final double minDecimalPart;
+    private Random random;
+    private ColorSupplier colorSupplier;
+    private DecimalFormat df;
+
+    public FigureSupplier() {
+        maxFigureSize = 100;
+        minDecimalPart = 0.001;
+        this.df = new DecimalFormat("#.###");
+        this.random = new Random();
+        this.colorSupplier = new ColorSupplier();
+    }
+
+    private double getRandomSize() {
+        String randomNumber = df.format(random.nextDouble() * maxFigureSize
+                + minDecimalPart);
+
+        return Double.parseDouble(randomNumber);
+    }
+
+    private String getRandomColor() {
+        return colorSupplier.getRandomColor();
+    }
+
+    public Figure getRandomFigure() {
+        int randomFigureIndex = random.nextInt(Figures.values().length);
```

Comment:
Instead of using `FigureName.values()`, you can just generate a random number from 0 to figures count.

Reply:
done